js/src/components/shipping-rate-section/minimum-order-card/minimum-order-card.js

```diff
@@ -14,75 +18,27 @@ import AppButtonModalTrigger from '.~/components/app-button-modal-trigger';
 import VerticalGapLayout from '.~/components/vertical-gap-layout';
 import isNonFreeFlatShippingRate from '.~/utils/isNonFreeFlatShippingRate';
 import MinimumOrderInputControl from './minimum-order-input-control';
-import AddMinimumOrderModal from './add-minimum-order-modal';
+import { AddMinimumOrderFormModal } from './minimum-order-form-modals';
 import groupShippingRatesByMethodFreeShippingThreshold from './groupShippingRatesByMethodFreeShippingThreshold';
+import {
+	getMinimumOrderAddHandler,
+	getMinimumOrderChangeHandler,
+	getMinimumOrderDeleteHandler,
+} from './handlers';
 import './minimum-order-card.scss';
 
 const MinimumOrderCard = ( props ) => {
-	const { value = [], onChange = noop } = props;
-	const nonZeroShippingRates = value.filter( isNonFreeFlatShippingRate );
-	const groups = groupShippingRatesByMethodFreeShippingThreshold(
-		nonZeroShippingRates
-	);
-	const countryOptions = nonZeroShippingRates.map(
-		( shippingRate ) => shippingRate.country
-	);
-
-	const handleEditChange = ( oldGroup ) => ( newGroup ) => {
-		const newValue = value.map( ( shippingRate ) => {
-			const newShippingRate = {
-				...shippingRate,
-				options: {
-					...shippingRate.options,
-				},
-			};
-
-			if ( newGroup.countries.includes( newShippingRate.country ) ) {
-				/**
-				 * Shipping rate's country exists in the new value countries,
-				 * so we just assign the new value threshold.
-				 */
-				newShippingRate.options.free_shipping_threshold =
-					newGroup.threshold;
-			} else if (
-				oldGroup.countries.includes( newShippingRate.country )
-			) {
-				/**
-				 * Shipping rate's country does not exist in the new value countries,
-				 * but it exists in the old value countries.
-				 * This means users removed the country in the edit modal,
-				 * so we set the threshold value to undefined.
-				 */
-				newShippingRate.options.free_shipping_threshold = undefined;
-			}
-
-			return newShippingRate;
-		} );
-
-		onChange( newValue );
-	};
-
-	const handleAddChange = ( newGroup ) => {
-		const newValue = value.map( ( shippingRate ) => {
-			const newShippingRate = {
-				...shippingRate,
-				options: {
-					...shippingRate.options,
-				},
-			};
-
-			if ( newGroup.countries.includes( newShippingRate.country ) ) {
-				newShippingRate.options.free_shipping_threshold =
-					newGroup.threshold;
-			}
-
-			return newShippingRate;
-		} );
-
-		onChange( newValue );
-	};
```

Comment:
Also, those function does not have much "event handling" functionality. All they do is translate API to API, then forward the result to an actual handler. So they are rather API translating proxy/glue-code, than "handlers". 

If we treat them as such, it would be easier to use, name, and test them.

Reply:
Also, this translation from `MinimumOrderGroup` to `ShippingRate` and back, seems like a wasted effort to me. `MinimumOrderGroup` type is created and used on the frontend, so we could translate back and work, writing additional functions like the ones above when we could work on the original data structure in all components.

Therefore, I'd rather avoid increasing the about of found around that translation, separate just the translation to the side, then eventually remove the need for it.

Reply:
While doing https://github.com/woocommerce/google-listings-and-ads/commit/f682f9e2fdb7b20c88804535fe4d80f5891a9f47, I realized that add and delete handlers are actually the change handlers - respectively without `oldGroup`, and without `newGroup`. So I'd roll them to one function

Reply:
See https://github.com/woocommerce/google-listings-and-ads/pull/1499 for full proposal, feel free to merge here any subset of its commits.

Reply:
@tomalec , thanks for the review and for your PR #1499, the PR looks good and I like it! 😃 👍 I left my review on your PR in https://github.com/woocommerce/google-listings-and-ads/pull/1499#pullrequestreview-974077665, and I think we can have both PRs merged into the `develop` branch, let me know how you like to proceed. 🙂